src/Renderer/ThreeExtented/GlobeControls.js

```diff
@@ -958,13 +1108,22 @@ function GlobeControls(camera, domElement, engine) {
         keyS = false;
     };
 
+    // Callback launched when player is stopped
+    this.stopAnimation = function () {
```

Comment:
If it is to be called each time an animation is finished, please consider having this callback as a parameter when constructing the `AnimationPlayer`. This way you won't need to add
```js
player.play().then(() => this.stopAnimation())
```
Or something similar each time. Then this callback would be called each time you call `this.resolve` in AnimationPlayer.

Btw, promises are not made for recurring events. They are made for one-time resolvable event, for which they are great. But as soon as your need can be expressed as

> I want to do A every time B happens

They are poor candidates. Callbacks are still the right solution for this.